### PR TITLE
feat(lexer) Introduce a user-defined lexer to support comments

### LINKED ISSUE #2: Support comments in the text parser
e.g.

```
;; a whole line comment
```

and

```
(as ;inline comment; any 0)
```

### PR DESCRIPTION
**Pull request by Hywan**
Fix #2.

This PR is a set of 4 patches:

  1. Introduce a user-defined lexer, that provides the exact same features than the auto-generated one,
  2. Replace the skip tokens algorithm based on `str.trim_start()` by a regex,
  3. Support inline comment,
  4. Support line comment.

I find unfortunate that token definitions (and their sibling regexes) must be duplicated in the code. I wasn't able to find a better way to do that by looking at  various existing LALRPOP usages.

I reckon the generated `grammar.rs` code is smaller with the user-defined lexer. The compilation time is still super slow though (see #13) and I wonder whether [nom](https://github.com/Geal/nom/) could be a better fit in this context.

Anyhow, Web IDL comments are now supported :-).

### CODE REVIEW

File: crates/text-parser/src/grammar.lalrpop [hunk 1]
```diff
@@ -350,3 +347,84 @@ Unsigned: u32 = <s:r"[0-9]+"> => u32::from_str(s).unwrap();
...
+        "view" => Token::View,
+        r"[0-9]+" => Token::Unsigned(<&'input str>),
+        r"[a-zA-Z$][a-zA-Z0-9$_]*" => Token::Identifier(<&'input str>),
+        r#""(([^\\"]|\\.)*)""# => Token::QuotedString(<&'input str>),
+    }
+}
+
+// Local Variables:
+// mode: rust
+// End:
```
Comment (fitzgen) [inline crates/text-parser/src/grammar.lalrpop]: Nitpick: Let's not put editor-specific comment pragmas in here.

**Review by fitzgen — COMMENTED** [review]
Looks great -- thanks @Hywan!

Just one nitpick below and this is good to merge.

File: crates/text-parser/src/grammar.lalrpop (hunk 1 above)
Comment (Hywan) [reply crates/text-parser/src/grammar.lalrpop]: Sure!

File: crates/text-parser/src/lexer.rs [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+                        "^unsigned long" => Token::UnsignedLong,
+                        "^unsigned short" => Token::UnsignedShort,
+                        "^utf8\\-cstr" => Token::Utf8CStr,
+                        "^utf8\\-str" => Token::Utf8Str,
+                        "^v128" => Token::V128,
+                        "^view" => Token::View,
+                        r"^([0-9]+)" => Token::Unsigned(result),
+                        r"^([a-zA-Z$][a-zA-Z0-9$_]*)" => Token::Identifier(result),
+                        r#"^"(([^\\"]|\\.)*)""# => Token::QuotedString(result),
+                        _ => panic!("oh no"),
```
Comment (Hywan) [inline crates/text-parser/src/lexer.rs]: I'll update that to an `unreachable!`.

**Comment by fitzgen** [comment]
Thanks!

**Comment by Hywan** [comment]
A pleasure :-)

**Pull request merged**